### DISCUSSION TITLE
Upgrade already installed software (installed via msi package)

### DISCUSSION THREAD

**Discussion by MalteKiefer** [microsoft/winget-cli#2481] [General]
Hello,

how can I upgrade already installed software (like Firefox, that was installed via MSI package) without first reinstalled it via winget?

Because currently it looks like this:

![image](https://user-images.githubusercontent.com/59220985/187218028-cf44ba8b-a5f7-4501-ac6e-43ed8c1f2d19.png)

And I have to do a `winget install Mozilla.Firefox` before it is upgraded by winget

**Comment by denelon** [comment]
@MalteKiefer Firefox has a large number of installers and doesn't yet have the matching "AppsAndFeatures" entries in the manifest to help with upgrade. If it's installed by winget, then a separate entry is created to help with the linking until that happens.

**Reply by MalteKiefer** [reply]
So how can I fix this, so that winget can update all these packages?

**Reply by denelon** [reply]
I would try uninstalling them and then reinstalling them with winget. I will also call out that if "upgrade" does an uninstall previous, you may need to pass in the "--locale" argument, otherwise winget will default to the OS locale for packages.
